master/internal/api_command.go

```diff
@@ -31,36 +34,122 @@ type protoCommandParams struct {
 	Config       *pstruct.Struct
 	Files        []*utilv1.File
 	Data         []byte
+	MustZeroSlot bool
+	Preview      bool
+}
+
+func (a *apiServer) makeFullCommandSpec(
+	configBytes []byte, templateName *string, mustBeZeroSlot bool,
+) (*model.CommandConfig, *tasks.TaskSpec, error) {
+	resources := model.ParseJustResources(configBytes)
+	taskSpec := a.m.makeTaskSpec(resources.ResourcePool, resources.Slots)
+	config := command.DefaultConfig(&taskSpec.TaskContainerDefaults)
+	if templateName != nil && *templateName != "" {
+		template, err := a.m.db.TemplateByName(*templateName)
+		if err != nil {
+			return nil, nil, errors.Wrapf(err, "failed to find template: %s", *templateName)
+		}
+		if err := yaml.Unmarshal(template.Config, &config); err != nil {
+			return nil, nil, errors.Wrapf(err, "failed to unmarshal template: %s", *templateName)
+		}
+	}
+
+	if len(configBytes) != 0 {
+		dec := json.NewDecoder(bytes.NewBuffer(configBytes))
+		dec.DisallowUnknownFields()
+
+		if err := dec.Decode(&config); err != nil {
+			return nil, nil, errors.Wrapf(
+				err,
+				"unable to parse the config in the parameters: %s",
+				string(configBytes),
+			)
+		}
+	}
+
+	// mustBeZeroSlot indicates that this type of command may never use more than
+	// zero slots (as of Jan 2021, this is only Tensorboards). This is important
```

Comment:
Instead of putting a date here, can we put a Determined version?

Reply:
This is a comment that is moved from other places in the codebase. I can't really find the exact version to which it belongs.  Anyways, this should be refactored later. I'd not worry about it